From 7d70dfe1831ddd37131dd123d0d1063f61e1dbf2 Mon Sep 17 00:00:00 2001
From: Hendrik Borghorst <hborghor@amazon.de>
Date: Fri, 2 Oct 2020 11:12:17 +0200
Subject: [PATCH] efi/x86: Disable disk-based non-volatile variable store

EDK2 has the nice option to have a disk-based store of the non-volatile
variables. We definitely don't want the guest firmware to write on
customer's disks.

Signed-off-by: Hendrik Borghorst <hborghor@amazon.de>
Reviewed-by: Evgeny Iakovlev <eyakovl@amazon.de>
Reviewed-by: Alexander Graf (AWS) <graf@amazon.de>

diff --git a/OvmfPkg/Library/PlatformBootManagerLib/BdsPlatform.c b/OvmfPkg/Library/PlatformBootManagerLib/BdsPlatform.c
index 98f6f07341..b9a86cba51 100644
--- a/OvmfPkg/Library/PlatformBootManagerLib/BdsPlatform.c
+++ b/OvmfPkg/Library/PlatformBootManagerLib/BdsPlatform.c
@@ -1664,12 +1664,6 @@ PlatformBootManagerAfterConsole (
       "PlatformBdsPolicyBehavior: not restoring NvVars "
       "from disk since flash variables appear to be supported.\n"
       ));
-  } else {
-    //
-    // Try to restore variables from the hard disk early so
-    // they can be used for the other BDS connect operations.
-    //
-    PlatformBdsRestoreNvVarsFromHardDisk ();
   }
 
   //
